extensions/positron-python/python_files/posit/positron/lsp.py

```diff
@@ -35,18 +34,13 @@ def on_comm_open(self, comm: BaseComm, msg: Dict[str, Any]) -> None:
 
         # Parse the host and port from the comm open message
         data = msg["content"]["data"]
-        client_address = data.get("client_address", None)
+        client_address = data.get("ip_address", None)
         if client_address is None:
-            logger.warning(f"No client_address in LSP comm open message: {msg}")
-            return
-
-        host, port = self._split_address(client_address)
-        if host is None or port is None:
-            logger.warning(f"Could not parse host and port from client address: {client_address}")
+            logger.warning(f"No ip_address in LSP comm open message: {msg}")
             return
 
         # Start the language server thread
-        POSITRON.start(lsp_host=host, lsp_port=port, shell=self._kernel.shell, comm=comm)
+        POSITRON.start(lsp_host=client_address, lsp_port=0, shell=self._kernel.shell, comm=comm)
```

Comment:
If we own the `start()` method and `start_tcp()`, then I would remove `lsp_port` from here as an argument entirely. I think it only adds to the confusion if you think you can specify your own port.

I'd only supply a port of 0 in one place, in `start_tcp()` in the `create_server()` call, and maybe with a comment about what 0 means there!

Reply:
Agreed, I suggest checking if we can leave the `port` parameter out of `create_server` as well. The default behavior if `port` is `None` might be what we want.

Reply:
I looked at the docs and I do think `None` is the same as `0`. A comment about why we aren't providing a port still seems like it would be useful above the `create_server()` call.